cosmoz-sse.js

```diff
@@ -1,230 +1,87 @@
-// @license Copyright (C) 2019 Neovici AB - Apache 2 License
-import { PolymerElement } from '@polymer/polymer/polymer-element';
-
-const SSEHandlers = {
-	text: data => data,
-	json: data => JSON.parse(data)
-};
-
-/**
-	`<cosmoz-sse>` is an element that allows subscribing to server-sent events.
-
-	@customElement
-	@polymer
-	@demo demo/index.html
-*/
-class CosmozSSE extends PolymerElement {
-	static get is() {
-		return 'cosmoz-sse';
-	}
-
-	static get properties() {
-		return {
-			_configuration: {
-				type: Object,
-				computed: '_computeConfiguration(withCredentials)'
-			},
-			/**
-			 * The SSE endpoint.
-			 */
-			url: {
-				type: String
-			},
-
-			/**
-			 * The named events you wish to subscribe to.
-			 */
-			events: {
-				type: Array,
-				value: [],
-				observer: '_onEventsChange'
+// @license Copyright (C) 2020 Neovici AB - Apache 2 License
+import {
+	component, useEffect, useMemo, useState
+} from 'haunted';
+
+const getJson = input => {
+		try {
+			return JSON.parse(input);
+		} catch (e) {
+			// eslint-disable-next-line no-console
+			console.error('cant parse JSON:', input);
+		}
+	},
+	SSEHandlers = {
+		text: data => data,
+		json: data => getJson(data)
+	},
+
+	// eslint-disable-next-line max-lines-per-function
+	CosmozSSE = function ({
+		events, withCredentials = false, handleAs = 'json', url
+	}) {
+		const fireEvent = (type, eventInit) => {
+				this.dispatchEvent(new CustomEvent(type, eventInit));
 			},
-
-			/**
-			 * How to parse the event data.
-			 * @type {('json'|'text')}
-			 */
-			handleAs: {
-				type: String,
-				value: 'json'
+			configuration = useMemo(() => ({
```

Comment:
I wouldn't bother with memoizing this. Just drop `configuration` and close over `withCredentials` directly in the effect that creates the EventSource.

Reply:
Even though it's not _necessary_ I think it adds better readability and documentation since it matches the spec?

Reply:
:+1: